substrate-archive-backend/src/util.rs

```diff
@@ -118,7 +117,7 @@ pub mod meta_keys {
 }
 
 pub fn read_header<Block: BlockT>(
-    db: &ReadOnlyDatabase,
+    db: Arc<dyn ReadOnlyDatabaseTrait>,
```

Comment:
AFAIU the introduced `Arc<dyn ReadOnlyDatabaseTrait>` seems needless, you should be able to have `&dyn ReadOnlyDatabaseTrait` and use `Arc::deref ` i.e, `&*self.db`